### DISCUSSION TITLE
Custom Antenna Pattern and LoRa Simulations

### DISCUSSION THREAD

**Discussion by ygtrz** [NVlabs/sionna#521] [Q&A] [answered]
Hello,

I have two question about the problems I have encountered using Sionna:

1) I am trying to define a custom antenna pattern and visualize it using Sionna. I have the given information of the antenna pattern in the format of: Zenith-azimuth angles and the corresponding absolute gain and axial ratio values in terms of dB. I found that I should define a callable and refer to it while defining my antenna. However, I could not understand the relation between Absolute Gain & Axial Ratio and Zenith & Azimuth Patterns. How should i transform the parameters?

2) I am trying to perform LoRa simulations in a ray tracing environment, but I could not find any defined function about it in the documentation. Is there any reference or advice how should I conduct LoRa simulations in a ray tracing environment?

Thank you for your time and consideration.

**Comment by jhoydis** [comment, marked as answer]
Hi @ygtrz,

1) Please read the EM Primer to understand how antennas are modeled in Sionna RT: https://nvlabs.github.io/sionna/em_primer.html#equation-c

If you have discrete values describing your antenna patter, you would need to interpolate it somehow.  Nearest neighbor interpolation could be a good starting point.

2) There is no specific support for LoRa in Sionna. However, ray tracing is agnostic to any standard as it models the wireless propagation channel.
